parcels/particleset.py

```diff
@@ -382,6 +382,45 @@ def show(self, **kwargs):
             print('Plot saved to '+savefile+'.png')
             plt.close()
 
+    def density(self, field=None, particle_val=None, relative=False, area_scale=True):
+        lons = [p.lon for p in self.particles]
+        lats = [p.lat for p in self.particles]
+        if field is not None:
+            # Kick out particles that are not within the limits of our density field
+            half_lon = (field.lon[1] - field.lon[0])/2
+            half_lat = (field.lat[1] - field.lat[0])/2
+            dparticles = (lons > (np.min(field.lon)-half_lon)) * (lons < (np.max(field.lon)+half_lon)) * \
+                         (lats > (np.min(field.lat)-half_lat)) * (lats < (np.max(field.lat)+half_lat))
+            dparticles = np.where(dparticles)[0]
+        else:
+            field = self.grid.U
+            dparticles = range(len(self.particles))
+        Density = np.zeros((field.lon.size, field.lat.size), dtype=np.float32)
+
+        # For each particle, find closest vertex in x and y and add 1 or val to the count
+        if particle_val is not None:
+            for p in dparticles:
+                Density[np.argmin(np.abs(lons[p] - field.lon)), np.argmin(np.abs(lats[p] - field.lat))] \
+                    += getattr(self.particles[p], particle_val)
+        else:
+            for p in dparticles:
+                nearest_lon = np.argmin(np.abs(lons[p] - field.lon))
+                nearest_lat = np.argmin(np.abs(lats[p] - field.lat))
```

Comment:
Ok, ideally we want to add the same cell-tracking to SciPy particle as we do in JIT, so that the grid cell is always known and doesn't need to be searched here. This will probably be much faster, since we can do this on a local search, whereas `np.argmin` traverses the entire lon/lat arrays _for each particle_, which is **very inefficient**. However, until we have cell tracking in Python mode, this is probably ok.